csv_to_geo_yaml

```diff
@@ -94,34 +93,44 @@ addresses.each do |address|
   end
 end
 
-locations.each do |location|
-  next if location[:results].any?
-  # we don't have any lat long data so let's search for it
-  search_string = [location[:address], location[:city], location[:state], location[:zip]].compact.join(", ")
-
-  geo_coder_results = Geocoder.search(search_string)
-
-  geo_coder_results.each do |geo_coder_result|
-    result = {}
-
-    geo_location = geo_coder_result.data['geometry']['location']
-    result[:address] = geo_coder_result.data['formatted_address']
-    result[:location_type] = geo_coder_result.data['geometry']['location_type']
-    result[:latitude] = geo_location['lat']
-    result[:longitude] = geo_location['lng']
-    result[:search_string] = search_string
-
-    location[:results] << result
+start_time = Time.now
+lookups_done = 0
+batches = locations.each_slice(100).to_a
+batches.each do |batch|
+  threads = []
+  batch.each do |location|
+    next if location[:results].any?
+    threads << Thread.new do
+      begin
+        # we don't have any lat long data so let's search for it
+        search_string = [location[:address], location[:city], location[:state], location[:zip]].compact.join(", ")
+
+        geo_coder_results = Geocoder.search(search_string)
+
+        geo_coder_results.each do |geo_coder_result|
+          result = {}
+
+          geo_location = geo_coder_result.data['geometry']['location']
+          result[:address] = geo_coder_result.data['formatted_address']
+          result[:location_type] = geo_coder_result.data['geometry']['location_type']
+          result[:latitude] = geo_location['lat']
+          result[:longitude] = geo_location['lng']
+          result[:search_string] = search_string
+
+          location[:results] << result
```

Comment:
At first glance, this looks problematic because of the `<<`, and I helped Drew change this to avoid using that... but after looking at the rest of the code I realized it was totally unnecessary. 

The important thing to note here is that `location` is always a unique single Hash element of the `locations` array, and that only one thread at a time is ever acting on each of those elements.  The fact that `<<` isn't atomic isn't actually relevant here, because `location` isn't shared state.  There is only a shared reference to it held in the `locations` array.  As long as another thread isn't going through that `locations` array and performing operations on the hashes within while these other threads are running, nothing untoward will happen.